harvester/oncokb.py

```diff
@@ -120,6 +131,71 @@ def convert(gene_data):
                                'oncokb': {'clinical': clinical}}
         yield feature_association
 
+    for biological in oncokb['biological']:
```

Comment:
After reviewing the data, it looks to me like the "biological" query in this PR is a superset of the "clinical" query above. As such, I don't think it's necessary that we make two requests on the API for each variant. Is there information that we'd miss by dropping the "clinical" query above and just keeping this more inclusive "biological" query?

Reply:
Subset is not quite accurate. There is definitely some overlap in the data that the two queries pull, when a variant is listed in both, but they both have data that's somewhat different. 

Specifically, for the "BCR-ABL1 Fusion" gene of "ABL1"...

Here is the clinical query:
```
{
    "cancerType": {
      "id": null,
      "code": "CML",
      "color": "LightSalmon",
      "name": "Chronic Myelogenous Leukemia",
      "mainType": {
        "id": null,
        "name": "Leukemia"
      },
      "tissue": "Blood",
      "children": {},
      "parent": "LEUK",
      "deprecated": false,
      "history": [],
      "links": null,
      "level": "Tertiary",
      "NCI": [
        "C3172"
      ],
      "UMLS": [
        "C0023470"
      ]
    },
    "variant": {
      "gene": {
        "entrezGeneId": 25,
        "hugoSymbol": "ABL1",
        "name": "ABL proto-oncogene 1, non-receptor tyrosine kinase",
        "oncogene": true,
        "curatedIsoform": "ENST00000318560",
        "curatedRefSeq": "NM_005157.4",
        "geneAliases": [
          "v-abl",
          "bcr/abl",
          "JTK7",
          "c-ABL1",
          "c-ABL",
          "ABL",
          "p150"
        ],
        "tsg": false
      },
      "consequence": {
        "term": "NA",
        "isGenerallyTruncating": false,
        "description": "NA"
      },
      "alteration": "BCR-ABL1 Fusion",
      "name": "BCR-ABL1 Fusion",
      "refResidues": null,
      "proteinStart": -1,
      "proteinEnd": 100000,
      "variantResidues": null
    },
    "level": "1",
    "drug": [
      "Nilotinib",
      "Imatinib",
      "Dasatnib"
    ],
    "drugPmids": [
      "11287973",
      "12637609",
      "20525993",
      "20525995",
      "22160483",
      "11287972",
      "23502220"
    ],
    "drugAbstracts": []
  }
```

And here is the biological query:
```
{
  "variant": {
    "gene": {
      "entrezGeneId": 25,
      "hugoSymbol": "ABL1",
      "name": "ABL proto-oncogene 1, non-receptor tyrosine kinase",
      "oncogene": true,
      "curatedIsoform": "ENST00000318560",
      "curatedRefSeq": "NM_005157.4",
      "geneAliases": [
        "v-abl",
        "bcr/abl",
        "JTK7",
        "c-ABL1",
        "c-ABL",
        "ABL",
        "p150"
      ],
      "tsg": false
    },
    "consequence": {
      "term": "NA",
      "isGenerallyTruncating": false,
      "description": "NA"
    },
    "alteration": "BCR-ABL1 Fusion",
    "name": "BCR-ABL1 Fusion",
    "refResidues": null,
    "proteinStart": -1,
    "proteinEnd": 100000,
    "variantResidues": null
  },
  "mutationEffect": "Gain-of-function",
  "mutationEffectPmids": [
    "11345193",
    "18851712"
  ],
  "mutationEffectAbstracts": [],
  "oncogenic": "Oncogenic",
  "oncogenicPmids": [],
  "oncogenicAbstracts": []
}
```

The clinical query has cancer and drug information that the biological query doesn't, which we would expect. But the biological query also has additional pubmed info on the ongenicity of the variant. 

Additionally, we've been working on the assumption that in order to assign a phenotype to any biological variants we pull we would use given oncogenic label. So in the case of the ABL1 example above, the phenotype of the first would be "Chronic Myelogenous Leukemia" and the phenotype of the second would be "Oncogenic."

The only way forward I can see with this issue, other than the current status, of multiple entries and phenotypes, is grab the pubmed ids from the biological query and add them to the pubmed ids from the clinical query. But that wouldn't be a completely depiction of the evidence, because those pubmed articles may have nothing to do with the cancer or drug referenced. Ultimately, I think the two queries and entries is better for this reason. Yes, it's a little stilted, but it's not like we don't already have variants with multiple entries for different phenotypes (cancers or drugs). 

Does this sufficiently answer your concern?

Reply:
I've spoken with Jeremy as well. With the addition of the sql silo, we agreed that spending time combining the clinical variants with the biological variants it's a good use of time, especially considering that we have other cases where the same variant is present in multiple documents, due to multiple cancers, etc.